### PR TITLE
Read body in the example server

### PR DESCRIPTION
**Pull request by nox**

### CODE REVIEW

**Comment by kornelski** [comment]
Whoa, `poll_closed` is a gotcha. Would it be possible to detect this error from `h2` somehow?

**Comment by nox** [comment]
I don't think so, the connection isn't gone or anything, it's just that no one is polling it anymore, ever.

**Comment by nox** [comment]
The docs could be improved and a proper future on which to await could be introduced though. `Connection::graceful_shutdown` exists but it still requires the connection to be polled until `Poll::Ready` afterwards, which is unfortunate IMO.

File: examples/server.rs [hunk 1]
```diff
@@ -27,18 +27,33 @@ async fn handle(socket: TcpStream) -> Result<(), Box<dyn Error + Send + Sync>> {
...
+    let (mut request, mut respond) = match connection.accept().await {
+        Some(result) => result?,
+        None => {
+            println!("Didn't get a request");
+            return Ok(());
+        }
+    };
+
+    tokio::spawn(async move {
+        let _ = futures_util::future::poll_fn(|cx| connection.poll_closed(cx)).await;
```
Comment (seanmonstar) [inline examples/server.rs:39]: I think this change will only allow a single request, right? I'd probably say we should change the example to just _always_ spawn a task to handle the request. Something like:

```rust
while let Some(result) = connection.accept().await {
    tokio::spawn(handle_request(result));
}
```

Comment (nox) [reply examples/server.rs:39]: That is true.

Comment (nox) [reply examples/server.rs]: I've made it use `tokio::spawn` instead.

Btw the window stuff seems weirdly handled:

```
<<<< recv 16384
<<<< recv 16384
<<<< recv 16384
<<<< recv 16383
<<<< recv 1
<<<< recv 16384
<<<< recv 16384
<<<< recv 1696
```

**Review by seanmonstar — APPROVED** [review]
Looks good to me! I dunno if you're still tweaking something, but merge when done :)

**Comment by nox** [comment]
> I dunno if you're still tweaking something

You know how it is, you think you are done and then the style lint complains.

**Pull request merged**